site/styles/_bootstrap4.scss

```diff
@@ -34,3 +34,93 @@
 @import 'bootstrap/scss/tooltip';
 @import 'bootstrap/scss/popover';
 @import 'bootstrap/scss/utilities';
+
+// The following override classes are to get around Bootstrap 4's use of Sass variables.
+// CSS themes will not work without these hacks.
+
+.btn-primary-override {
+  @extend .btn-primary;
+  border-color: $button-background;
```

Comment:
How does this work? I can see the Sass var picking up from the CSS vars in `_variables.scss`:

```scss
--primary-rgb: 53, 160, 168;
--primary: rgb(var(--primary-rgb));
$button-background: var(--primary) !default;
border-color: $button-background;
```

So when the Sass is first compiled, it will have CSS for `border-color: var(--primary)`? Allowing override for either `--primary` or the underlying `--primary-rgb`?

Reply:
- `--primary-rgb` is required for all the places where we use the `rgba` function.
- `--primary` is required for all the places where we (and Bootstrap) use colours directly without passing them through `rgba`.
- Those Bootstrap override classes are required for places where Bootstrap 'fails' silently while passing colours through functions.  In the case of the primary button, some colours would work and some would default back to Bootstrap colours.  I've extracted _all_ of the colours for the primary button however so it's easier to see the big picture in one place even though half of them don't actually need an override.

Does that make sense?

Reply:
Yeah that makes sense, but wasn't what I was asking. I was asking for the compiled output in `main.css`, if the CSS in there was like this:

```css
/* Variables decalared early in the file */
--primary-rgb: 53, 160, 168;
--primary: rgb(var(--primary-rgb));

/* ... and much later when they are used (compiled from Sass - border-color: $button-background;) */
border-color: var(--primary);
```

And if so, then if the model was to have `local.css` be able to override the vars `--primary-rgb` or `--primary`?

Reply:
Yeah `local.css` can override any of the `--` vars.